src/db.js

```diff
@@ -25,7 +25,7 @@ class Db {
       });
   }
 
-  async createParty(name) {
+  async createParty(name, callback) {
```

Comment:
does it make sense to return a promise instead of only having the async method?

If this were a promise you would be able to just say, 
```javascript
createParty(name).then(c => //do what you want to do next)
```
this is more readable from my POV. I'm not sure if you want to go down this path at this moment.

Reply:
This a tricky one, the async here is for the `await` when we set the party in the database and the callback function is to handle the changes, for example, every time a player joins or flees.

We can remove the `async` so the code looks like:
``` javascript
return this.db
  .collection('parties')
  .doc(id)
  .set(party)
  .then(() => {
    this.unsubscribeParty = this.subscribeParty(party.id, callback);
    return party;
  });
```

Instead of:
``` javascript
await this.db
  .collection('parties')
  .doc(id)
  .set(party);

this.unsubscribeParty = this.subscribeParty(party.id, callback);

return party;
```

I would like your opinion about this because the same structure is used in several methods.

Reply:
got it, my bad, by default the async methods return a promise when you don't await, so what I was suggesting is a workaround to avoid passing down the `callback` method. 

```javascript
createParty(name).then(p => callback());
```

I don't think you need the callback property if you don't await from the caller.